lib/pre/reqQueueDepth.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 Restify. All rights reserved.
+
+'use strict';
+
+var assert = require('assert-plus');
+
+
+///--- API
+
+
+/**
+ * capture request queue depth.
+ * @public
+ * @function createReqQueueDepth
+ * @param {Object} opts an options object
+ * @param {Object} opts.server a restify server
+ * @returns {Function}
+ */
+function createReqQueueDepth(opts) {
+
+    assert.object(opts, 'opts');
+    assert.object(opts.server, 'opts.server');
+
+    if (typeof opts.server.getReqQueueDepth === 'function') {
+        throw new Error('server object already has `getReqQueueDepth()` ' +
+                        'method!') ;
+    }
+
+    var QUEUE_LENGTH = 0;
+
+    // create method on server
+    opts.server.getReqQueueDepth = function getReqQueueDepth() {
+        return QUEUE_LENGTH;
+    };
+
+    return function reqQueueDepth(req, res, next) {
+
+        // increase request queue depth
+        QUEUE_LENGTH += 1;
+
+        function _pop() {
+            QUEUE_LENGTH -= 1;
+        }
+
+        // only one of the three below events will get fired per request.
+        // for happy case when the response is properly flushed
+        res.once('finish', _pop);
+        // for when client (or Zuul) kills the request, or gives up prematurely
+        res.once('close', _pop);
```

Comment:
doesn't this mean you'll be "leaking" the queue count. If you're incrementing but never decrementing cus u have uncaughts, this will keep going up?